### ISSUE TITLE
Statistical tests on Classification and Regression data and models

### ISSUE THREAD

**Issue by aliagowani** [pycaret/pycaret#1517]
A new feature is being implemented for the Timeseries module to evaluate relevant statistical tests on both pre and post models. It would be meaningful to implement this module throughout Pycaret and in particular with the Classification and Regression modules. 

Allow the user to call statistical tests for classification and regression modules (see the attached screenshot that was shared for the time series module).

While a user can generate these statistics by calling the get_config function, it would be much more efficient to generate these during the processing of the data and do what Pycaret excel at: remove or minimize redundant and tedious task when evaluating data and generating models. 

Note: please ensure the output of the statistical tests can be easily called so that it can be leveraged for outputs (reports, plots, etc.).

![1629039623061](https://user-images.githubusercontent.com/51864674/129662854-25ad9cb0-ed35-471c-a96e-f48977905485.jpg)

**Comment by moezali1** [comment]
@aliagowani That's a great idea. What kind of statistical tests do you normally perform on classification and regression problems? Anything you can share will help us create a requirement for this proposal.

**Comment by moezali1** [comment]
@ngupta23 Do you have any thoughts on this?

**Comment by aliagowani** [comment]
@moezali1 I have some thoughts and will share more details in few days. 

@ngupta23 would love to get your thoughts to keep things consistent with time series module as well.

**Comment by ngupta23** [comment]
For regression: 
I think the only test that would be applicable from the time series list is the normality test on the model residuals. This would give you the goodness of model fit. The rest of the tests in the list are specific to time series so they will not apply.

We could add a summary stats section (for target) along with that. 

For classification: I am not sure any of these would apply.

**Comment by aliagowani** [comment]
@ngupta23 I agree with you that this may only apply to regression models as there isn't much to add than what Pycaret does for classification models, such as, confusion matrix, adding custom metrics to models, etc. One thing to note is that it would be good to add some of output of the graphs (e.g.: confusion matrix) to global variables (e.g.: True Positives (TP), False Positives (FP), etc.)

For Regression models, I would look at statsmodel to see what can be incorporated (e.g.: standard error, p-value, confidence intervals, etc.). Since much of this has been applied to Time Series module, it should be reusable? 

![Screen Shot 2021-08-24 at 10 11 52 AM](https://user-images.githubusercontent.com/51864674/130642629-dbf29df4-e940-475e-bce2-393d9f178852.png)

**Comment by Ali-Flt** [comment]
Hi, 
I wanted to check if there is any updates on this rather important issue. Regression output could be considered irrelevant without knowing the uncertainty of each coefficient and uncertainty of the residuals so p-values, t-values and confidence intervals are a critical output need to be reported in the model evaluation phase.

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
